numpyro/mcmc.py

```diff
@@ -79,20 +78,16 @@ def init_kernel(init_samples,
         hmc_state = HMCState(vv_state.z, vv_state.z_grad, vv_state.potential_energy, 0, 0.,
                              wa_state.step_size, wa_state.inverse_mass_matrix, rng_hmc)
 
+        jwa_update = jit(wa_update)
         if run_warmup:
-            hmc_state, _ = jit(fori_loop, static_argnums=(2,))(0, num_warmup_steps, warmup_update,
-                                                               (hmc_state, wa_state))
+            for t in range(num_warmup_steps):
```

Comment:
Does this change the timing? If so, I would suggest doing this in a separate PR so that it is easier to figure out how the timings are affected by each change separately.

Reply:
I thought that it will help us avoid 2 time compiling. But I agree that it is better to do in a separate PR.